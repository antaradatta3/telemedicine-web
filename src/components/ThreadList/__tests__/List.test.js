import React from 'react';
import { mount } from 'enzyme';

import List from '../List';

describe('Component: List', () => {
  const props = {
    items: [
      {
        data: {
          id: '1',
          ups: 100,
          author: 'sample',
          downs: 0,
          replies: {
            data: {
              children: [],
            },
          },
        },
      },
      {
        data: {
          id: '2',
          ups: 100,
          author: 'sample',
          downs: 0,
          replies: {
            data: {
              children: [],
            },
          },
        },
      },
    ],
  };

  it('List matches snapshot', () => {
    const component = mount(<List {...props} />);
    expect(component).toMatchSnapshot();
  });
});
