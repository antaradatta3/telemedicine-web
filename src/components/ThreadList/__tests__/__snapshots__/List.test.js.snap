// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: List List matches snapshot 1`] = `
<List
  items={
    Array [
      Object {
        "data": Object {
          "author": "sample",
          "downs": 0,
          "id": "1",
          "replies": Object {
            "data": Object {
              "children": Array [],
            },
          },
          "ups": 100,
        },
      },
      Object {
        "data": Object {
          "author": "sample",
          "downs": 0,
          "id": "2",
          "replies": Object {
            "data": Object {
              "children": Array [],
            },
          },
          "ups": 100,
        },
      },
    ]
  }
>
  <div>
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        <a
          rel="noopener noreferrer"
          target="_blank"
        />
      </div>
      <div
        className="card-body"
      >
        <img
          alt="Avatar"
          height={60}
          src="https://www.redditstatic.com/avatars/avatar_default_04_D4E815.png"
          width={60}
        />
        <h5
          className="card-title"
        />
        <p
          className="card-text"
        >
          <Badge
            label="Comments"
            type="primary"
          >
            <span>
               
              <span
                className="badge badge-pill badge-primary"
              />
               
              Comments
                
            </span>
          </Badge>
          <Badge
            count={100}
            label="Ups"
            type="success"
          >
            <span>
               
              <span
                className="badge badge-pill badge-success"
              >
                100
              </span>
               
              Ups
                
            </span>
          </Badge>
          <Badge
            count={0}
            label="Downs"
            type="danger"
          >
            <span>
               
              <span
                className="badge badge-pill badge-danger"
              >
                0
              </span>
               
              Downs
                
            </span>
          </Badge>
          <Badge
            label="Score"
            type="info"
          >
            <span>
               
              <span
                className="badge badge-pill badge-info"
              />
               
              Score
                
            </span>
          </Badge>
        </p>
        <button
          className="btn btn-success btn-block"
          onClick={[Function]}
          type="button"
        >
          Open
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        <a
          rel="noopener noreferrer"
          target="_blank"
        />
      </div>
      <div
        className="card-body"
      >
        <img
          alt="Avatar"
          height={60}
          src="https://www.redditstatic.com/avatars/avatar_default_04_D4E815.png"
          width={60}
        />
        <h5
          className="card-title"
        />
        <p
          className="card-text"
        >
          <Badge
            label="Comments"
            type="primary"
          >
            <span>
               
              <span
                className="badge badge-pill badge-primary"
              />
               
              Comments
                
            </span>
          </Badge>
          <Badge
            count={100}
            label="Ups"
            type="success"
          >
            <span>
               
              <span
                className="badge badge-pill badge-success"
              >
                100
              </span>
               
              Ups
                
            </span>
          </Badge>
          <Badge
            count={0}
            label="Downs"
            type="danger"
          >
            <span>
               
              <span
                className="badge badge-pill badge-danger"
              >
                0
              </span>
               
              Downs
                
            </span>
          </Badge>
          <Badge
            label="Score"
            type="info"
          >
            <span>
               
              <span
                className="badge badge-pill badge-info"
              />
               
              Score
                
            </span>
          </Badge>
        </p>
        <button
          className="btn btn-success btn-block"
          onClick={[Function]}
          type="button"
        >
          Open
        </button>
      </div>
    </div>
  </div>
</List>
`;
